src/libs/actions/Report.js

```diff
@@ -795,6 +795,46 @@ function addAction(reportID, text, file) {
         .then(({reportAction}) => updateReportWithNewAction(reportID, reportAction));
 }
 
+/**
+ * Deletes a comment from the report, basically sets it as empty string
+ *
+ * @param {Number} reportID
+ * @param {Object} reportAction
+ */
+function deleteReportAction(reportID, reportAction) {
+    // Optimistic Response
+    const reportActionsToMerge = {};
+    const oldMessage = {...reportAction.message};
+    reportActionsToMerge[reportAction.sequenceNumber] = {
+        ...reportAction,
+        message: [
+            {
+                type: 'COMMENT',
+                html: '',
+                text: '',
+            },
+        ],
+    };
+
+    Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`, reportActionsToMerge);
+
+    // Mark the report as not having any unread items
+    API.Report_EditComment({
+        accountID: currentUserAccountID,
+        reportID,
+        reportActionID: reportAction.sequenceNumber,
```

Comment:
Something to keep in mind is that the `reportActionID` _is not_ the same as the sequence number. The `reportActionID` is actually the `ROWID` of the `reportAction` from the `reportActions` table in the database. So I think this what you want:

```suggestion
        reportActionID: reportAction.reportActionID,
```

Reply:
I tried using the `reportActionID` but that did not work, in fact this only worked when I used `sequenceNumber`

Reply:
cc @yuwenmemon, this should be `reportActionID` right? Any idea why it wouldn't be working for @sketchydroide

Reply:
I agree that I think it should be the action ID and I think figuring out why it doesn't work is a good route to take.